lib/services/database/database_service.dart

```diff
@@ -0,0 +1,239 @@
+import 'dart:async';
+
+import "package:collection/collection.dart";
+import 'package:mobileprism/services/database/album_data_entry.dart';
+import 'package:mobileprism/services/database/cross_table_entry.dart';
+import 'package:mobileprism/services/database/photo_data_entry.dart';
+import 'package:path/path.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:sqflite/sqflite.dart';
+
+const String photoDataTableName = "photo_data";
+const String albumDataTableName = "album_data";
+const String keyCrosstableName = "key_crosstable";
+const String photoTableCreationStrg = '''
+      CREATE TABLE IF NOT EXISTS $photoDataTableName(uid TEXT PRIMARY KEY, 
+      panorama INTEGER, width INTEGER, height INTEGER,
+      image_hash TEXT, image_quality TEXT, lat REAL, long REAL, 
+      timestamp INETEGER)
+    ''';
+const String albumTableCreationStrg = '''
+    CREATE TABLE IF NOT EXISTS $albumDataTableName(uid TEXT PRIMARY KEY, 
+    photo_uids TEXT, title TEXT, thumb_uid TEXT)
+  ''';
+const String keyCrosstableCreationStrg = '''
+    CREATE TABLE IF NOT EXISTS $keyCrosstableName(photo_uid Text, album_uid Text)
+  ''';
+// const String databaseConfigString = 'PRAGMA foreign_keys = ON';
+
+late final Database? _database;
+
+class DbNotOpenException implements Exception {}
+
+class FieldsMissingException implements Exception {}
+
+class DuplicateKeyException implements Exception {}
+
+class KeyNotFoundException implements Exception {}
```

Comment:
I prefer having an extra file for exceptions like `database_exceptions.dart`

Reply:
One file for each exception would be a bit overkill I think. But maybe one DbExceptions file.